### PR TITLE
feat(remote config): Add a settings page for remote config basics

### LINKED ISSUE #71128: [Remote Config] Implement /settings/remote-config page within Sentry based on api.md in #70805
https://github.com/getsentry/sentry/pull/70805/files

### PR DESCRIPTION
**Pull request by ryan953**
Create a new project settings page for Remote Config options. This is the first verison of the page, def not the last! We'll be changing the api that this page talks to, as well as the buttons and layout as we iterate.

The purpose of this version is to make it possible for people to save their configs.

SCR-20240603-hmxz

Known problems:
- the delete button works to delete the config, but it doesn't render toasts properly on the screen because the backend is returning a 503 for some reason. 
- you will be prevented from creating two 'features' with the same key, but the ui doesn't give feedback, needs a "this is a duplicate key" label to be added. 

Fixes https://github.com/getsentry/sentry/issues/71128

### CODE REVIEW

File: static/app/views/settings/project/projectRemoteConfig.tsx [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+  project: Project;
+  result: ReturnType<typeof useRemoteConfigSettings>['result'];
+}) {
+  const disabled = result.isError;
+
+  return (
+    <FieldGroup
+      label={t('Delete Remote Config')}
+      help={tct(
+        'If you want to start over you can delete the remote config for [projectName].',
```
Comment (michellewzhang) [inline static/app/views/settings/project/projectRemoteConfig.tsx:345]:
```suggestion
        'If you want to start over, you can delete the remote config for [projectName].',
```

File: static/app/routes.tsx [hunk 2]
```diff
@@ -548,6 +548,13 @@ function buildRoutes() {
         name={t('Replays')}
         component={make(() => import('sentry/views/settings/project/projectReplays'))}
       />
+      <Route
+        path="remote-config/"
+        name={t('Replays')}
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/routes.tsx:553]: d'oh

[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/routes.tsx:553]:
```suggestion
        name={t('Remote Config')}
```

File: static/app/views/settings/project/projectRemoteConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+  options: RemoteConfigOptions;
+}) {
+  return (
+    <Fragment>
+      <FieldGroup
+        id="sample_rate"
+        label={t('Sample Rate')}
+        help={t(
+          'Configures the sample rate for error events, in the range of 0.0 to 1.0. The default is 1.0 which means that 100% of error events are sent. If set to 0.1 only 10% of error events will be sent. Events are picked randomly.'
+        )}
```
[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/views/settings/project/projectRemoteConfig.tsx:160]: This was copy+pasted from the docs: https://docs.sentry.io/platforms/javascript/configuration/options/#sample-rate

[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/views/settings/project/projectRemoteConfig.tsx:160]: Fix in docs: https://github.com/getsentry/sentry-docs/pull/10248

**Comment by c298lee** [comment]
A toast saying unable to save remote config always shows up even though the values do seem to save
image

**Comment by ryan953** [comment]
> A toast saying unable to save remote config always shows up even though the values do seem to save

@c298lee Yeah, this is a known problem. the backend is always timing out and returning 503 for some reason. It's beta stuff so it'll get sorted out in the next iteration. After that I'm confident this FE code will work because it's the same structure as the handleSave call.

**Review by c298lee — APPROVED** [review]

**Pull request merged**